### PR TITLE
Add xlookup and logarithm operators

### PR DESCRIPTION
**Pull request by lthorpedo**
- Breaks out our custom extension operators into a separate file (instead of adding onto the already 1000+ line json_logic.ex file)
- Adds xlookup and logarithm operations/operators.

### CODE REVIEW

**Comment by lthorpedo** [comment]
going to add tests for the xlookup and logarithm and exponent ops.

File: mix.lock [hunk 1]
```diff
@@ -13,6 +13,7 @@
   "makeup": {:hex, :makeup, "1.1.0", "6b67c8bc2882a6b6a445859952a602afc1a41c2e08379ca057c0f525366fc3ca", [:mix], [{:nimble_parsec, "~> 1.2.2 or ~> 1.3", [hex: :nimble_parsec, repo: "hexpm", optional: false]}], "hexpm", "0a45ed501f4a8897f580eabf99a2e5234ea3e75a4373c8a52824f6e873be57a6"},
   "makeup_elixir": {:hex, :makeup_elixir, "0.16.0", "f8c570a0d33f8039513fbccaf7108c5d750f47d8defd44088371191b76492b0b", [:mix], [{:makeup, "~> 1.0", [hex: :makeup, repo: "hexpm", optional: false]}, {:nimble_parsec, "~> 1.2.3", [hex: :nimble_parsec, repo: "hexpm", optional: false]}], "hexpm", "28b2cbdc13960a46ae9a8858c4bebdec3c9a6d7b4b9e7f4ed1502f8159f338e7"},
   "makeup_erlang": {:hex, :makeup_erlang, "0.1.1", "3fcb7f09eb9d98dc4d208f49cc955a34218fc41ff6b84df7c75b3e6e533cc65f", [:mix], [{:makeup, "~> 1.0", [hex: :makeup, repo: "hexpm", optional: false]}], "hexpm", "174d0809e98a4ef0b3309256cbf97101c6ec01c4ab0b23e926a9e17df2077cbb"},
+  "math": {:hex, :math, "0.7.0", "12af548c3892abf939a2e242216c3e7cbfb65b9b2fe0d872d05c6fb609f8127b", [:mix], [], "hexpm", "7987af97a0c6b58ad9db43eb5252a49fc1dfe1f6d98f17da9282e297f594ebc2"},
```
Comment (lthorpedo) [inline mix.lock]: don't need the math module, or to update this dependency

File: README.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
 # JsonLogic with Exponent Operator
```
Comment (tom-anglepoint) [inline README.md]: missed a spot?

Comment (lthorpedo) [reply README.md]: Thanks!

**Pull request merged**